bfg: add page (#5984)
tldr-pages/tldr: pages/common/bfg.md

# bfg

> Remove large files or passwords from Git history like git-filter-branch.
> Note: if your repository is connected to a remote, you will need to force push to it.
> More information: <https://rtyley.github.io/bfg-repo-cleaner/>.

> - Remove a file with sensitive data and leave the latest commit untouched:

Reviewer: "And" makes it sound as though it's doing an action, but it's really just an omission.
```suggestion
- Remove a file with sensitive data but leave the latest commit untouched:
```

Author: I agree!

# bfg

> > Remove large files or passwords from Git history like git-filter-branch

Reviewer 2: ```suggestion
> Remove large files or passwords from Git history like git-filter-branch.
```

As soon as the workflow is approved, the bot will flag this

> More information: <https://rtyley.github.io/bfg-repo-cleaner/>.

- Remove a file with sensitive data and leave the latest commit untouched:

> `bfg --delete-files {{FILE-WITH-SENSITIVE-DATA}}`

Reviewer 2: ```suggestion
`bfg --delete-files {{file_with_sensitive_data}}`
```

> - Replace all text listed in passwords.txt wherever it can be found in the repository's history:
> 
> `bfg --replace-text {{passwords.txt}}`

Reviewer 2: ```suggestion
- Replace all text listed in text.txt wherever it can be found in the repository's history:

`bfg --replace-text {{text.txt}}`
```

> - Replace all text listed in passwords.txt wherever it can be found in the repository's history:
> 
> `bfg --replace-text {{passwords.txt}}`

Reviewer 2: ```suggestion
- Remove all text mentioned in text.txt wherever it can be found in the repository's history:

`bfg --replace-text {{text.txt}}`
```

Reviewer: I'd rather something a bit more generic, e.g.
```suggestion
- Remove all text mentioned in the specified file wherever it can be found in the repository's history:

`bfg --replace-text {{path/to/file.txt}}`
```

- After the sensitive data is removed, you must force push your changes to GitHub:

> `git push --force`

Reviewer 2: We normally don't reference other commands. We may add something to one of the other command, like these changes are only commited, not pushed to origin

Reviewer 2: @bl-ue what's your opinion on this

Reviewer 2: Also this needs to not be Github, just origin since gitlab and bitbucket do exist

Reviewer: There's a few problems with this example:
1. As @CleanMachine1 points out, it's a different command other than the one that we're documenting, but that's of the least concern
2. The command assumes that the repository is connected to a remote, which it may, in fact, not be
3. GitHub isn't the only remote repository service ;)

I suggest moving this information to a new line in the page description, something like
```
> Note: you will need to force-push if your repository is connected to a remote.
```

Author: Thank you so much for your detailed feedback! I'll make these changes within a day or two when I get some time 😊

> `git push --force`

Reviewer 2: ```suggestion
`git push --force`

```
Need a new line at the end \n
